src/Models/Hooks/useRuntimeSceneData.ts

```diff
@@ -63,116 +71,265 @@ export const useRuntimeSceneData = (
      * update the colored meshes ids based on run expressions in behaviors against the returned ADT twin property data
      *  */
     useEffect(() => {
-        if (sceneData?.adapterResult?.result?.data) {
-            const sceneVisuals = deepCopy(
-                sceneData.adapterResult.result.data.sceneVisuals
-            );
+        // Keep old method intact in case flighting is turned off
+        if (showVisualRulesFeature) {
+            if (sceneData?.adapterResult?.result?.data) {
+                const sceneVisuals = deepCopy(
+                    sceneData.adapterResult.result.data.sceneVisuals
+                );
+
+                if (selectedLayerIds) {
+                    const behaviorIdsInSelectedLayers = ViewerConfigUtility.getBehaviorIdsInSelectedLayers(
+                        scenesConfig,
+                        [...selectedLayerIds],
+                        sceneId
+                    );
+
+                    // Apply layer filtering to behaviors - splice out behaviors not in selected layers
+                    sceneVisuals.forEach((sv) => {
+                        const filteredBehaviors = sv.behaviors.filter((b) =>
+                            behaviorIdsInSelectedLayers.includes(b.id)
+                        );
+                        sv.behaviors = filteredBehaviors;
+                    });
+                }
+
+                const twinIds = new Set<string>();
+                const badgeVisuals: Array<{
+                    sceneVisual: SceneVisual;
+                    sceneViewBadge: SceneViewBadge;
+                }> = [];
+                // for each scene visual retrieve the colored mesh ids and update it in the scene visual
+                // if they are triggered by the element's behaviors and currently active
+                sceneVisuals.forEach((sceneVisual) => {
+                    sceneVisual.coloredMeshItems = [];
+
+                    for (const twinId in sceneVisual.twins) {
+                        twinIds.add(sceneVisual.twins[twinId].$dtId);
```

Comment:
nit: we have a constant for the $dtid value now. Can you use that from constants please?

Reply:
This is actually getting the value of the twin's $dtId, what do you mean by using the one from constants?